public/pages/CreateSnapshotPolicy/components/CronSchedule/CronSchedule.tsx

```diff
@@ -0,0 +1,173 @@
+import moment from "moment-timezone";
+import { CRON_EXPRESSION_DOCUMENTATION_URL } from "../../../../utils/constants";
+
+interface CronScheduleProps {
+  frequencyType: string;
+  onChangeFrequencyType: (e: ChangeEvent<HTMLSelectElement>) => void;
+  cronExpression: string;
+  onCronExpressionChange: (expression: string) => void;
+  timezone?: string;
+  onChangeTimezone?: (e: ChangeEvent<HTMLSelectElement>) => void;
+}
+
+const CronSchedule = ({
+  frequencyType,
+  onChangeFrequencyType,
+  cronExpression,
+  onCronExpressionChange,
+  timezone,
+  onChangeTimezone,
+}: CronScheduleProps) => {
+  const { minute: initMin, hour: initHour, dayOfWeek: initWeek, dayOfMonth: initMonth } = parseCronExpression(cronExpression);
+
+  const [minute, setMinute] = useState(initMin);
+  const [hour, setHour] = useState(initHour);
+  const [dayOfWeek, setWeek] = useState(initWeek);
+  const [dayOfMonth, setMonth] = useState(initMonth);
+
+  useEffect(() => {
+    changeCron();
+  }, [minute, hour, dayOfWeek, dayOfMonth]);
+
+  const changeCron = (input?: any) => {
+    let cronParts = { hour, minute, dayOfWeek, dayOfMonth, frequencyType };
+    cronParts = { ...cronParts, ...input };
+    const expression = buildCronExpression(cronParts, cronExpression);
+    console.log(`sm dev built expression ${expression}`);
+    onCronExpressionChange(expression);
+  };
+
+  function onDayOfWeekChange(dayOfWeek: string) {
+    setWeek(dayOfWeek);
+    // changeCron({ dayOfWeek });
+  }
+
+  function onDayOfMonthChange(dayOfMonth: number) {
+    setMonth(dayOfMonth);
+    // changeCron({ dayOfMonth });
+  }
+
+  function onStartTimeChange(date: moment.Moment) {
+    const minute = date.minute();
+    const hour = date.hour();
+    setMinute(minute);
+    setHour(hour);
+    // changeCron({ minute, hour });
+  }
+
+  function onTypeChange(e: ChangeEvent<HTMLSelectElement>) {
+    const frequencyType = e.target.value;
+    onChangeFrequencyType(e);
+    changeCron({ frequencyType });
+  }
+
+  const dayOfWeekCheckbox = (day: string, checkedDay: string) => (
+    <EuiFlexItem key={day} grow={false} style={{ marginRight: "0px" }}>
+      <EuiCheckbox id={day} label={day} checked={checkedDay === day} onChange={(e) => onDayOfWeekChange(day)} compressed />
+    </EuiFlexItem>
+  );
+
+  let additionalContent;
+  if (frequencyType === "weekly") {
+    additionalContent = <EuiFlexGroup>{WEEK_DAYS.map((d) => dayOfWeekCheckbox(d, dayOfWeek))}</EuiFlexGroup>;
+  }
+  if (frequencyType === "monthly") {
+    additionalContent = (
+      <EuiFlexGroup gutterSize="m">
+        <EuiFlexItem>
+          <EuiSelect options={[{ value: "day", text: "Day" }]} defaultValue="Day" />
+        </EuiFlexItem>
+        <EuiFlexItem>
+          <EuiFieldNumber
+            value={dayOfMonth}
+            onChange={(e) => {
+              console.log(`sm dev change day of month ${parseInt(e.target.value)}`);
+              onDayOfMonthChange(parseInt(e.target.value));
+            }}
+          />
```

Comment:
This field currently allows users to enter negative numbers and numbers greater than 31.  The `EuiFieldNumber` element has `min` and `max` fields that could be used to help control the input here. The UI does display a toast with the following info when attempting to submit a policy with a day number that's not between 1 and 31.
```
Failed to create snapshot policy: {"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"Failed to parse cron expression. Value 38293 not in range [1, 31]"}],"type":"illegal_argument_exception","reason":"Failed to parse cron expression. Value 38293 not in range [1, 31]","caused_by":{"type":"illegal_argument_exception","reason":"Value 38293 not in range [1, 31]"}},"status":400}

Reply:
Add min max